bigframes/remote_function.py

```diff
@@ -202,10 +202,22 @@ def create_bq_remote_function(
             OPTIONS (
               endpoint = "{endpoint}"
             )"""
+
         logger.info(f"Creating BQ remote function: {create_function_ddl}")
+
+        # Make sure the dataset exists
+        dataset = bigquery.Dataset(
+            bigquery.DatasetReference.from_string(
+                self._bq_dataset, default_project=self._gcp_project_id
+            )
+        )
+        dataset.location = self._bq_location
+        self._bq_client.create_dataset(dataset, exists_ok=True)
```

Comment:
What happens if the dataset exists while the user doesn't have dataset creation permission?

Reply:
Let me test that. Btw, this is happening in the remote function flow, where a lot more permissions are required, so requiring  `bigquery.datasets.create` shouldn't bother us

Reply:
> What happens if the dataset exists while the user doesn't have dataset creation permission?

tested that they still get the error